amy/templates/emails/scheduled_email_detail.html

```diff
@@ -0,0 +1,163 @@
+    <a class="btn btn-warning disabled" disabled href="#">Cancel</a>
+  </div>
+</div>
+{% endif %}
+
+<table class="table table-striped">
+  <tr>
+    <th width="20%">ID:</th>
+    <td>{{ scheduled_email.id }}</td>
+  </tr>
+  <tr>
+    <th>State:</th>
+    <td><strong>{{ scheduled_email.get_state_display }}</strong></td>
+  </tr>
+  <tr>
+    <th>Created at:</th>
+    <td>{{ scheduled_email.created_at }}</td>
+  </tr>
+  <tr>
+    <th>Last updated at:</th>
+    <td>{{ scheduled_email.last_updated_at }}</td>
+  </tr>
+  <tr>
+    <th>Scheduled to run:</th>
+    <td>{{ scheduled_email.scheduled_at }}</td>
+  </tr>
+  <tr>
+    <th>From:</th>
+    <td>{{ scheduled_email.from_header }}</td>
+  </tr>
+  <tr>
+    <th>To:</th>
+    <td>
+      <ul>
+        {% for email in scheduled_email.to_header %}
+          <li>{{ email }}</li>
+        {% endfor %}
+      </ul>
+    </td>
+  </tr>
+  <tr>
+    <th>Reply-To:</th>
+    <td>{{ scheduled_email.reply_to_header|default:"&mdash;" }}</td>
+  </tr>
+  <tr>
+    <th>CC:</th>
+    <td>
+    {% if scheduled_email.cc_header %}
+      <ul>
+      {% for email in scheduled_email.cc_header %}
+        <li>{{ email }}</li>
+      {% endfor %}
+      </ul>
+    {% else %}
+    &mdash;
+    {% endif %}
+    </td>
+  </tr>
+  <tr>
+    <th>BCC:</th>
+    <td>
+    {% if scheduled_email.bcc_header %}
+      <ul>
+      {% for email in scheduled_email.bcc_header %}
+        <li>{{ email }}</li>
+      {% endfor %}
+      </ul>
+    {% else %}
+    &mdash;
+    {% endif %}
+    </td>
+  </tr>
+  <tr>
+    <th>Subject:</th>
+    <td><code>{{ scheduled_email.subject }}</code></td>
+  </tr>
+  <tr>
+    <th>Body: <br><small>(Markdown; converts to HTML when sent)</small></th>
+    <td>
+      <pre>{{ scheduled_email.body }}</pre>
+    </td>
+  </tr>
+  <tr>
+    <th>Template:</th>
+    <td><a href="{{ scheduled_email.template.get_absolute_url }}">{{ scheduled_email.template }}</a></td>
+  </tr>
+  <tr>
+    <th>Related object:</th>
+    <td>
+      {% if scheduled_email.generic_relation and scheduled_email.generic_relation.get_absolute_url %}
+      <a href="{{ scheduled_email.generic_relation.get_absolute_url }}">{{ scheduled_email.generic_relation }}</a>
+      {% elif scheduled_email.generic_relation %}
+      {{ scheduled_email.generic_relation }}
+      {% else %}
+      &mdash;
+      {% endif %}
+    </td>
+  </tr>
+  <tr>
+    <th>History:<br><small>(newest to oldest)</small></th>
+    <td>
+      <table class="table">
+        <tr>
+          <th>Timestamp</th>
+          <th>State before</th>
+          <th>State after</th>
+          <th>Details</th>
+          <th>ID</th>
+        </tr>
+
+        {% for log in log_entries %}
+        <tr>
+          <td>{{ log.created_at }}</td>
+          <td>{{ log.state_before }}</td>
+          <td>{{ log.state_after }}</td>
+          <td><code>{{ log.details }}</code></td>
+          <td>{{ log.id }}</td>
+        </tr>
+        {% endfor %}
+      </table>
+    </td>
```

Comment:
This log is a great improvement on what we currently have! The human-readable `details` make it easy to understand.

From an accessibility perspective, I don't think having this table nested inside a bigger table is a good idea. Could you move it to be after the other table and use "History" as the [table caption](https://www.w3.org/WAI/tutorials/tables/caption-summary/)? I think you can use heading markup inside the `<caption>` element if needed.

Reply:
Thanks, I wasn't aware of the issues of the nested table. I will move it out.